### ISSUE TITLE
IDISP011 false positive

### ISSUE THREAD

**Issue by sungam3r** [DotNetAnalyzers/IDisposableAnalyzers#246]
```c#
 public IProducer CreateProducer(string queueName, Action<ProducerOptions> setupAction = null)
        {
            GuardDisposed();

            var producer = _connection.CreateProducer(queueName, setupAction);
            producer.AddDisposeAction(() =>
            {
                lock (_producersLock)
                    _producers.Remove(producer);
            });
            lock (_producersLock)
                _producers.Add(producer);
            return producer; // IDISP011	Don't return disposed instance.
        }


 public interface IProducer : IDisposable
    {
        Task SendMessageAsync(Message message);
    }
```

**Comment by sungam3r** [comment]
```c#
 public static Process Spawn()
        {
            var process = new Process();
            process.StartInfo.CreateNoWindow = false;
            process.StartInfo.UseShellExecute = true;

            if (IsNetCoreApp)
            {
                process.StartInfo.FileName = "dotnet";
                process.StartInfo.Arguments = Assembly.GetEntryAssembly().Location;
            }
            else
            {
                process.StartInfo.FileName = Assembly.GetEntryAssembly().Location;
            }

            process.Start();
            return process;  // IDISP011	Don't return disposed instance.
        }
```

**Comment by JohanLarsson** [comment]
I did not have all the types for the first repro, the second was already fixed. Assuming both are fixed, please reopen if first example is not fixed in next release.

**Comment by JohanLarsson** [comment]
https://www.nuget.org/packages/IDisposableAnalyzers/3.4.5

**Comment by sungam3r** [comment]
fixed!
